apps/web/components/booking/BookingListItem.tsx

```diff
@@ -127,7 +126,11 @@ function BookingListItem(booking: BookingItemProps) {
         booking.listingStatus === "recurring" && booking.recurringEventId !== null
           ? t("cancel_all_remaining")
           : t("cancel"),
-      href: `/cancel/${booking.uid}`,
+      href: `/cancel/${booking.uid}${
+        booking.listingStatus === "recurring" && booking.recurringEventId !== null
+          ? "?allRemainingBookings=true"
```

Comment:
When cancelling we need to let the UI and the API know if the intention is to cancel all remaining bookings or just that booking instance.